### ISSUE TITLE
determine ordering of fields for item page and facet and any missing fields

### ISSUE THREAD

**Issue by yulgit1** [ycba-cia/blacklight-collections2#18]

**Comment by flapka** [comment]
The current ordering of fields might be improved in multiple ways.

RB would especially like to move the "contributors" field to the end of the record. Some items in our collection have *many* contributors; listing them all near the top makes it harder to immediately see the more essential metadata. See for example Letters to Richard Regrave: http://10.5.96.187:3000/catalog/orbis:12646943

**Comment by edgartdata** [comment]
@flapka @KraigBinkowski @yulgit1 
Ed and I would like to propose this order for the facets:

- Collection
- Creator
- Date
- Period
- Classification
- Medium
- Dimensions
- Subject Terms
- Place Represented 
- People Represented (could we change it to People Associated?)
- Credit Line
- Currently on View
- Image Available
- Item Rights

**Comment by edgartdata** [comment]
@yulgit1 Would it be possible to have facets be dynamic, i.e. have some appear according to the Collection selected?
For example, it would be helpful if the 5 facets of Work Type, Collective Title, Creator Gender, Genre, and Frame Style were to appear in the facet panel when the Prints & Drawings, Paintings & Sculpture, and Frames Collections are selected.

**Comment by flapka** [comment]
@edgartdata @yulgit1 

Does Blacklight already accomplish this? See, for example, the facets displayed when limited to RB collections:
http://10.5.96.187:3000/?f%5Bcollection_ss%5D%5B%5D=Rare+Books+and+Manuscripts

It omits the facets for which we have no data (e.g. collective title, creator gender ...) -- whereas these same facets appear when looking at other collections. Functionally, this seems to be what @edgartdata is asking for.

**Comment by flapka** [comment]
@KraigBinkowski wrote (via email):

I would think that subject terms might be a bit higher in the facet order – I feel this will be a highly used facet. But I see how it fits nicely with  places and people so I’m not sure what the solution would be. Currently on View I also see as a particularly important facet and if we are ordering them according to how we feel they will be used, then this one might be higher also (unless there is an easy way to see if a given object is on view or not, without using the facet).

**Comment by flapka** [comment]
@edgartdata @KraigBinkowski 

I think the facet order proposed is generally okay -- especially since some of the facets will disappear when limited to RB, REF, or IA collections. I'm inclined to put dimensions farther down (above credit line?).

**Comment by yulgit1** [comment]
Like Francis said, I think what Emmanuelle is asking for is the current behavior.  Otherwise awaiting consensus before implementing.

**Comment by edgartdata** [comment]
@EdwardTown1 It might be helpful to discuss this at our meeting with Elisabeth and Matthew on April 9?

**Comment by EdwardTown1** [comment]
Definitely. Sounds good

**Comment by edgartdata** [comment]
@EdwardTown1 and @edgartdata to suggest a tentative order of fields for detailed object view as well as for facets.
@KraigBinkowski @flapka RB and Ref Lib will aim to be as close as possible knowing that some RB and Ref Lib fields will be just for RB and Ref Lib.

**Comment by edgartdata** [comment]
@flapka @KraigBinkowski @EdwardTown1 @yulgit1 

Suggested order for facets and detailed record fields:

(Please remove Frame style altogether from the facets.)

Facets: 
•	Collection
•	Creator
•	Creator gender 
•	Collective title
•	Period 
•	Date
•	Medium
•	Height
•	Width 
•	Classification
•	Work type
•	Genre
•	Subject Terms
•	Subject Period
•	Place Represented
•	People Represented (could we change it to Associated People?)
•	Currently on View
•	Image Available
•	Item Rights
•	Credit Line

Detailed record field order:
•	Creator
•	Title
•	Date
•	Medium
•	Dimensions
•	Inscription/mark/lettering
•	Credit line
•	Accession number
•	Classification
•	Collection
•	Subject terms 
•	Place represented 
•	Associated people 
•	publications
•	Link 
•	Export

**Comment by flapka** [comment]
This is my stab at RB’s preferences, as a starting point. I defer to Elisabeth in our discussion on the 23rd.

For Facets, the list is in Emmanuelle's order; the numbers to the right indicate the RB preferences. 
1.	Collection – RB 1
2.	Creator – 2 
3.	Creator gender
4.	Collective title
5.	Period – 5 
6.	Date – 4 
7.	Medium
8.	Height
9.	Width
10.	Classification – 3 
11.	Work type – 11*
12.	Genre – 12* 
13.	Subject Terms – 7
14.	Subject Period – 8 
15.	Place Represented – 9 
16.	People Represented (could we change it to Associated People?) – 10 
17.	Currently on View
18.	Image Available – 6
19.	Item Rights – 13 
20.	Credit Line – 14 

Certain fields don’t have library/archive equivalents (not yet at least): gender, collective title, medium, height, width, currently on view. I think RB will have data to map to Work type and Genre, but I’m still exploring how best to map it.

**Comment by flapka** [comment]
@edgartdata @KraigBinkowski @EdwardTown1 @yulgit1

For detailed record field order, RB's preferences:

1.	Creator
2.	Title
3.	Alternative title
4.	Edition
5.	Published / created
6.	Physical description
7.	Collection
8.	Call number
9.	Credit line
10.	Aeon request* [icon?]
11.	Full Orbis Record [link]
12.	Related content [856 field links]
13.	Classification
14.	Scale [cartographic]
15.	Notes
16.	Contents
17.	Subject Terms
18.	Form / Genre
19.	Contributors
20.	Cite As

No need to display, for library records:
•	Subject period
•	RDF (as Export option – for now at least)
•	Cite as -- should appear only if there is data mapped from a MARC 524 field

**Comment by yulgit1** [comment]
@flapka @KraigBinkowski Do you want the language facet? If so where should it go in Emmanuelle/Ed's list above?

**Comment by flapka** [comment]
@KraigBinkowski @yulgit1 
This might be more important to Ref than it is to RB, since Ref has more foreign language material (maybe?).

I could see it slotting right above Subject Terms. Or right above Currently on View.

**Comment by KraigBinkowski** [comment]
We have a fair amount of foreign language materials but I don't think this facet would be used very much at all.  --  I think it might even go further down the list - maybe even below currently on view. 
I would not be apposed to doing away with this facet to clean up the list a bit if others agree.

**Comment by edgartdata** [comment]
The art collection doesn't use the language facet.

**Comment by yulgit1** [comment]
Implemented above with:
https://github.com/ycba-cia/blacklight-collections2/commit/d9e1472d865f56ddb84697499f920e0a32979af8

@flapka @edgartdata @EdwardTown1 @KraigBinkowski 
Please check that the ordering and inclusion of facets and detail page fields are to your liking

**Comment by flapka** [comment]
@edgartdata @EdwardTown1 @KraigBinkowski @yulgit1 

YUL's Quicksearch (Blacklight) includes a "Recently added" facet, which I find useful, occasionally. Would we want anything like it?

![image](https://user-images.githubusercontent.com/7634899/56763630-e0fa2a00-6770-11e9-9d94-c77bfebe5ba0.png)

**Comment by edgartdata** [comment]
@flapka Would such a facet be for records newly added to the catalog? or for new acquisitions? I would prefer a new acquisition facet but @EdwardTown1 and others might differ.

**Comment by flapka** [comment]
@edgartdata That's a good question. For RB and REF, newly added to the catalog would be easier. The date the record was created is automatically stored in the MARC record. No such automation for date of acquisition. For RB, we record an acquisition number in a part of the record that's not visible to the public. That number corresponds to an entry in a separate database of acquisitions (where you'll find the actual date of acquisition).

**Comment by KraigBinkowski** [comment]
For Ref the majority would be the latest publication date, though we certainly also acquire materials that aren't new. So using the date the record is created is best. I'm not sure what would happen when we add new holdings to a record that already exists? These could all still be called "New Acquisitions" though, even if they are newly added to the catalog.

**Comment by edgartdata** [comment]
Kraig, that’s what I’m thinking too. Surely at this point, all records newly added to our respective online catalogs are for new acquisitions, or are they not? And if they are not, I question the value of new records vs. new acquisitions for our patrons.

**Comment by flapka** [comment]
@yulgit1 The ordering looks correct. 

Two simple fixes:
* don't display "Cite As" -- it's unnecessary, and we have so few records with such data
* don't display RDF link -- we don't have such information for library collections (not yet at least)

**Comment by flapka** [comment]
> Kraig, that’s what I’m thinking too. Surely at this point, all records newly added to our respective online catalogs are for new acquisitions, or are they not? And if they are not, I question the value of new records vs. new acquisitions for our patrons.

This is mostly true, but there are exceptions. We may, for example, acquire a large collection, and initially create a single broad collection-level description for it. It may be a while later (sometimes years) before we have the time to create numerous individual records for every item within that collection. So, if the facet in "Blacklight" is based on the date of record creation for library material, and if we call the facet something like "new acquisitions," there is the potential to offer misleading information.

**Comment by edgartdata** [comment]
Valid case for the art collection as well when we catalog all the prints in a portfolio for example. Can we consider creating 2 different facets then? one for records newly released, and the other for new acquisitions?

**Comment by yulgit1** [comment]
@flapka - fixed the simple fixes above

@flapka @edgartdata @EdwardTown1 @KraigBinkowski - what is the consensus on the new facet(s) for newly released or acquired? Do these fields currently exist?

**Comment by edgartdata** [comment]
@EdwardTown1 @flapka @KraigBinkowski Do you think 2 facets would be overkill? One for records newly released, and the other for new acquisitions?

**Comment by KraigBinkowski** [comment]
Yes - overkill.

**Comment by edgartdata** [comment]
Then I vote for a new acquisitions facet.

**Comment by EdwardTown1** [comment]
Entirely indifferent here. If the consensus holds that this will be useful
then I'd vote in favour.

**Comment by flapka** [comment]
A real date of acquisition isn't currently in the data (or easily made available) for library collections. So my preference would be for date of addition. 

Would it be so bad to blur the meaning of the facet by using a label such as "Recent additions," allowing art collections to use date of acquisition if so inclined and library collections to use the date the record was created?

**Comment by flapka** [comment]
> @flapka - fixed the simple fixes above

@yulgit1 Looks great. Thanks!

**Comment by edgartdata** [comment]
I really like the label "Recent additions" to cover both types of records!

**Comment by edgartdata** [comment]
Eric will filter on TMS accession numbers to capture all the records created in the last year. The rule for the art collection would be dependent on the B#.

The rule for RB would be to grab all records created in the last 12 months from the day.

**Comment by flapka** [comment]
From Kalee, at YUL:

Our ‘Recently Added’ sort is based on a solr field we have built called ‘acq_dt’, or acquisition date.

We build the ‘acq_dt’ field contents on extract from Voyager.  We pull the acquisition date from line items in our invoice records.  For those records that don’t have associated invoice records (usually older records), we use the bib record create date.  We then add a local 907$a with the date to the MARC record and ingest it as ‘acq_dt’ in solr. 

So basically we’re focusing on when the records were added to our collection according to our invoice records.  Using the 008 field as you suggest might be an easier way to go about it, if that data is consistently present for you guys.

Here’s the code in our solr.rb file:


config.add_facet_field 'acq_dt',
--

**Comment by edgartdata** [comment]
@yulgit1 @flapka What is the status of this issue for RB today?

@yulgit1 Can we test the addition of a new facet called 'Recent additions' based on the filtering logic for the art collection? and add RB data when it is available please?

**Comment by flapka** [comment]
@yulgit1 @edgartdata 

Eric, the coded record date information for library material will come from the [008 field](http://www.loc.gov/marc/bibliographic/bd008a.html), positions 00-05 ("date entered on file").

I hope you're in the mood for dark humor: this coded date is six characters, because the first two digits of the year are omitted (thanks MARC!). For example, in the following item, the record date is 100330 (2010 March 30): 

https://libapp.library.yale.edu/OAI_BAC/src/OAIOrbisTool.jsp?verb=GetRecord&identifier=oai:orbis.library.yale.edu:9292982&metadataPrefix=marc21

Fortunately, we weren't creating MARC records in 1918, 1919, 1920, and so on. Eric, do you think you'll be able to work with this? Do you need anything from me?

**Comment by yulgit1** [comment]
Added 'New Additions' Facet
resolved by:
https://git.yale.edu/ermadmix/ycba_xslts/commit/1f83cc09093666f6bc8627fee93f5e585d36aab8
and:
https://git.yale.edu/ermadmix/camel_collections2/commit/e75cd2d33deb6d24cf17674e8b62a605739a5bc5
and: https://github.com/ycba-cia/blacklight-collections2/commit/b5549d68c32945a2cbd4d694b2f77095099b86fb

**Comment by flapka** [comment]
The MARC XSLT looks like it will give us correct data for at least another 40 years. Thanks.
